variables.mk

```diff
@@ -175,7 +175,8 @@ override SCALA_BUILDTOOL_DEPS += $(SBT_THIN_CLIENT_TIMESTAMP)
 SBT_CLIENT_FLAG = --client
 endif
 
-SBT ?= java $(JAVA_TOOL_OPTIONS) -jar $(ROCKETCHIP_DIR)/sbt-launch.jar $(SBT_OPTS) $(SBT_CLIENT_FLAG)
+SBT_BIN ?= java $(JAVA_TOOL_OPTIONS) -jar $(ROCKETCHIP_DIR)/sbt-launch.jar $(SBT_OPTS)
+SBT ?= $(SBT_BIN) $(SBT_CLIENT_FLAG)
```

Comment:
Should `SBT_OPTS` be part of `SBT_BIN` or `SBT`? I would expect `SBT_OPTS` to be the same for various settings of `SBT_BIN`

Reply:
Hmm this also drops `JAVA_TOOL_OPTIONS` when `SBT_BIN` is custom set. Is that intended?

Reply:
When using the SBT script the variables in the `SBT_OPTS` makefile variable are passed to the script in two ways:
1. the `.sbtopts` file in the top-level
2. the env. variable `SBT_OPTS` (which is assumed to be set by users)

As for JAVA_TOOL_OPTIONS, I need to double check...

Reply:
I wonder if Chipyard should just automatically download SBT (its just a .jar and a bash script) and use it by default...
then we wouldn't have to have all this makefile stuff around it.

Reply:
Ok. Cleaned this up a bit but I can confirm that doing `export var` exports the variable to the recipe. Therefore, `JAVA_TOOL_OPTIONS` is passed to `java` no matter what.